lib/govuk_publishing_components/presenters/machine_readable/article_schema.rb

```diff
@@ -100,18 +102,22 @@ def linked_page(step_by_step)
       end
 
       def has_part
-        return {} unless collection_pages.any?
+        return {} unless collection_pages("documents").any?
         {
-            "hasPart" => collection_pages.map { |document| HasPartSchema.new(document).structured_data }
+            "hasPart" => collection_pages("documents").map { |document| HasPartSchema.new(document).structured_data }
         }
       end
 
-      def collection_pages
-        @pages ||= fetch_collection_pages
+      def document_collections
```

Comment:
If, as per my suggestion for line 74, you call this method twice, you could memoize this as you've done with `step_by_step_schemas` and `fetch_step_by_step_schemas`

Reply:
👍